### PR TITLE
Add Map Layers component to Scenario Details

### PR DESCRIPTION
**Pull request by lattemilk**
* Added map layer panel in the Map Layers tab by reusing most of the code in set-priorities (thanks Hana!)
* Note: Not sharing the component because of differences in form logic, but can optimize later if it's worth the effort!
* Plan map shows/hides layers when visibility icon is toggled
* XS bug fix on layer visibility (not hiding the layer when untoggled)
* The subset of tests should be the same in the map layer component as in set-priorities, so didn't bother writing tests

# UI
Screenshot 2023-02-27 at 6 22 12 PM

### CODE REVIEW

File: src/interface/src/app/plan/scenario-details/map-layers/map-layers.component.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      .subscribe((conditionsConfig) => {
+        this.datasource.data = this.conditionsConfigToPriorityData(
+          conditionsConfig!
+        );
+      });
+  }
+
+  ngOnChanges() {
+    if (this.plan) {
+      console.log('map-layers ngOnChanges', { plan: this.plan });
```
Comment (thequeenofspades) [inline src/interface/src/app/plan/scenario-details/map-layers/map-layers.component.ts:74]: remove?

File: src/interface/src/app/plan/scenario-details/scenario-details.component.ts [hunk 2]
```diff
@@ -74,7 +74,25 @@ export class ScenarioDetailsComponent implements OnInit {
...
+      tap((scenario) => console.log(scenario)),
       takeUntil(this.destroy$)
     );
   }
+
+  private getPlan() {
+    return this.planService.planState$.pipe(
+      map((state) => {
+        if (state.currentPlanId) {
+          console.log(state.all[state.currentPlanId]);
```
Comment (thequeenofspades) [inline src/interface/src/app/plan/scenario-details/scenario-details.component.ts:86]: remove?

File: src/interface/src/app/plan/scenario-details/scenario-details.component.ts [hunk 3]
```diff
@@ -74,7 +74,25 @@ export class ScenarioDetailsComponent implements OnInit {
         });
         return of(null);
       }),
+      tap((scenario) => console.log(scenario)),
```
Comment (thequeenofspades) [inline src/interface/src/app/plan/scenario-details/scenario-details.component.ts:77]: remove?

**Review by thequeenofspades — APPROVED** [review]

File: src/interface/src/app/plan/scenario-details/map-layers/map-layers.component.ts (hunk 1 above)
Comment (lattemilk) [reply src/interface/src/app/plan/scenario-details/map-layers/map-layers.component.ts]: woops, done!

File: src/interface/src/app/plan/scenario-details/scenario-details.component.ts (hunk 2 above)
Comment (lattemilk) [reply src/interface/src/app/plan/scenario-details/scenario-details.component.ts]: done

File: src/interface/src/app/plan/scenario-details/scenario-details.component.ts (hunk 3 above)
Comment (lattemilk) [reply src/interface/src/app/plan/scenario-details/scenario-details.component.ts]: done!

**Pull request merged**